app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.java

```diff
@@ -0,0 +1,544 @@
+          (dialog, id) -> openMobileDataActivity());
+      builder.setNegativeButton((android.R.string.no), (dialog, id) -> startHotspotDialog());
+      builder.setTitle(this.getString(R.string.mobile_data_enabled));
+      builder.setMessage(
+          this.getString(R.string.mobile_data_message) + "\n" + this.getString(
+              R.string.mobile_data_message_confirmation)
+      );
+      builder.setCancelable(false);
+      AlertDialog dialog = builder.create();
+      dialog.show();
+    }
+  }
+
+  @Override public void serverStarted(@NonNull String ip) {
+    this.ip = ip;
+    serverTextView.setText(getString(R.string.server_started_message, this.ip));
+    startServerButton.setText(getString(R.string.stop_server_label));
+    startServerButton.setBackgroundColor(getResources().getColor(R.color.stopServer));
+    isServerStarted = true;
+  }
+
+  @Override public void serverStopped() {
+    serverTextView.setText(getString(R.string.server_textview_default_message));
+    startServerButton.setText(getString(R.string.start_server_label));
+    startServerButton.setBackgroundColor(getResources().getColor(R.color.greenTick));
+    isServerStarted = false;
+  }
+
+  @Override public void serverFailed() {
+    Toast.makeText(this, R.string.server_failed_message, Toast.LENGTH_LONG).show();
+  }
+
+  @Override public void hotspotTurnedOn(@NonNull WifiConfiguration wifiConfiguration) {
+
+    //Show an alert dialog for hotspot details
+    AlertDialog.Builder builder = new AlertDialog.Builder(this, dialogStyle());
+
+    ProgressDialog progressDialog =
+        ProgressDialog.show(this, getString(R.string.progress_dialog_starting_server), "",
+            true);
+    progressDialog.show();
+
+    builder.setPositiveButton(android.R.string.ok, (dialog, id) -> {
+      final Handler handler = new Handler();
+      handler.postDelayed(() -> {
+        progressDialog.dismiss();
+        startService(ACTION_START_SERVER);
+      }, 2000);
+    });
+
+    builder.setTitle(this.getString(R.string.hotspot_turned_on));
+    builder.setMessage(
+        this.getString(R.string.hotspot_details_message) + "\n" + this.getString(
+            R.string.hotspot_ssid_label) + " " + wifiConfiguration.SSID + "\n" + this.getString(
+            R.string.hotspot_pass_label) + " " + wifiConfiguration.preSharedKey);
+
+    builder.setCancelable(false);
+    AlertDialog dialog = builder.create();
+    dialog.show();
+
+    //setupServer();
+  }
+
+  void setupWifiSettingsIntent() {
+    final Intent intent = new Intent(Intent.ACTION_MAIN, null);
+    intent.addCategory(Intent.CATEGORY_LAUNCHER);
+    final ComponentName cn =
+        new ComponentName("com.android.settings", "com.android.settings.TetherSettings");
+    intent.setComponent(cn);
+    intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+    startActivity(intent);
+  }
+
+  @Override public void hotspotFailed() {
+    //Show a dialog to turn off default hotspot
```

Comment:
If you need a comment to explain it then maybe it has a bad name. `KiwixDialog.TurnOffHotSpot`
tells me that it is a dialog and it is meant to turn off a hotspot and when I click on it... well it does this `setupWifiSettingsIntent();` but there is a separate comment for that

Reply:
Fixed. 396526af9a6b605b44cde1ea8aea5732695a5892